tree/tree/inc/ROOT/RFriendInfo.hxx

```diff
@@ -58,13 +63,19 @@ struct RFriendInfo {
     */
    std::vector<std::vector<std::int64_t>> fNEntriesPerTreePerFriend;
 
-   void AddFriend(const std::string &treeName, const std::string &fileNameGlob, const std::string &alias = "");
+   /**
+      Information on the friend's TTreeIndexes.
+   */
+   std::vector<TVirtualIndex *> fTreeIndexInfos;
```

Comment:
Pass-by comment. It's scary IMO to see a vector of raw pointers in a new public interface. I see the point in having it to avoid copies, but I wonder if this is the only way. If it is, I would spend some wording in the docs to discuss the lifetime and ownership of the `TVirtualIndex *` inside the vector.

Reply:
Mmmh thank you for pointing this out. The raw pointers are non-owning and point to the tree indexes in the original friends (which in the RDF/TTreeProcessorMT case must have lifetimes at least as long as the event loop, so this saves us a copy for the local case). For the distributed case, if we stream the RFriendInfo to the tasks, I think we'll leak the tree indexes.

Reply:
> For the distributed case, if we stream the RFriendInfo to the tasks, I think we'll leak the tree indexes.

Yes I was also thinking about that. Currently we don't support TTreeIndex in distributed mode, but this would make it simply impossible to support I believe.

Reply:
switched to a `vector<unique_ptr<TVirtualIndex>>` although it makes the patch a bit more comlpicated

Reply:
> For the distributed case, if we stream the RFriendInfo to the tasks, I think we'll leak the tree indexes.

You we have had something like
```
std::vector<TVirtualIndex *> fTreeIndexInfos;
bool fIsOwning = false;
```
and a read rule setting `fIsOwning` to `true` when reading back and a condition deletion of the content.

This would avoid having to duplicate the `TTreeIndex` needlessly on the main process.

Reply:
I don't think the extra complication (I'm thinking mostly about the read rule and its testing) is worth it: the only drawback of the current solution is that it requires one extra copy of the tree index: if we ever need that (N+1)-th of the RAM back we can revise, but for now the priority is fixing https://github.com/root-project/root/issues/12260 .